executor/src/witgen/sequence_iterator.rs

```diff
@@ -176,6 +176,13 @@ impl ProcessingSequenceIterator {
             Self::Cached(it) => it.len() > 0,
         }
     }
+
+    pub fn is_cached(&self) -> bool {
```

Comment:
Is it clear what aspects or what kind of cache is meant here?

Reply:
I'd say so, no? A `ProcessingSequenceIterator` is just an enum and this tests for one of the two possible variants?